Wrappers/Python/cil/framework/framework.py

```diff
@@ -4207,14 +4210,19 @@ def allocate(self, value=0, **kwargs):
                 seed = kwargs.get('seed', None)
                 if seed is not None:
                     numpy.random.seed(seed) 
-                out.fill(numpy.random.random_sample(self.shape))
+                if numpy.iscomplexobj(out.array):
+                    out.fill(numpy.random.random_sample(self.shape) + 1.j*numpy.random.random_sample(self.shape))
+                else:
+                    out.fill(numpy.random.random_sample(self.shape))
             elif value == VectorGeometry.RANDOM_INT:
                 seed = kwargs.get('seed', None)
                 if seed is not None:
                     numpy.random.seed(seed)
-                max_value = kwargs.get('max_value', 100)
-                r = numpy.random.randint(max_value,size=self.shape, dtype=numpy.int32)
-                out.fill(numpy.asarray(r, dtype=self.dtype))             
+                max_value = kwargs.get('max_value', 100)                            
+                if numpy.iscomplexobj(out.array):
```

Comment:
Is this check also relevant for RANDOM_INT in AcquisitionGeometry.allocate?

Reply:
Good spot!